arch/wasm32/wasm.js

```diff
@@ -932,77 +932,82 @@ var ffi = (function() {
 if (arguments.length < 1)
   throw new Error('Expected at least one wasm module to load.');
 
-function load_wasm(file_path) {
-  // TODO this should be split up in load, check dependencies, and then resolve
-  // dependencies. That would make it easier to do lazy loading. We could do
-  // this by catching load exceptions + adding to ffi and trying again, but
-  // we're talking silly there: modules should just tell us what they want.
-  const buf = (typeof readbuffer === 'function')
-    ? new Uint8Array(readbuffer(file_path))
-    : read(file_path, 'binary');
-  instance = new WebAssembly.Instance(new WebAssembly.Module(buf), ffi)
-  heap = instance.exports.memory.buffer;
-  heap_uint8 = new Uint8Array(heap);
-  heap_size_bytes = heap.byteLength;
-  return instance;
-}
-
-// Load modules in reverse, adding their exports to the ffi object.
-// This is analogous to how the linker resolves symbols: the later modules
-// export symbols used by the earlier modules, and allow shadowing.
-// Note that all modules, as well as the main module, share a heap.
-var modules = {};
-for (var i = arguments.length - 1; i > 0; --i) {
-  var path = arguments[i];
-  modules[i] = load_wasm(path);
-  for (var f in modules[i]) {
-    // TODO wasm modules don't have hasOwnProperty. They probably should.
-    //      The code should look like:
-    //      if (modules[i].hasOwnProperty(f) &&
-    //          modules[i][f] instanceof Function)
-    ffi[f] = modules[i][f];
+if (arguments[0] == '--dump-ffi-symbols') {
+  for (var name in ffi["env"])
+    print(name)
```

Comment:
This js file already used print() directly to do all its output.  I don't know why we want to support both.. at least not as part of this change.